### PR TITLE
Nonblocking and gss_iov

### PR DESCRIPTION
**Pull request by ffilz**

### CODE REVIEW

**Comment by ffilz** [comment]
Don't merge this yet, found more issues

**Comment by dang** [comment]
It needs to be squashed into sane commits before merging, anyway.

File: src/xdr_ioq.c
Comment (dang) [inline src/xdr_ioq.c:703]: Note that this is only safe because uvq is the first filed in xdr_ioq_uv.  Otherwise NULL would result in a negative pointer value.

File: src/xdr_ioq.c
Comment (dang) [inline src/xdr_ioq.c:332]: This should be either unconditional or removed...

**Review by dang — COMMENTED** [review]
Minor nit.  Otherwise looks good.

File: src/xdr_ioq.c
Comment (ffilz) [reply src/xdr_ioq.c:332]: I thought about removing it completely, it was really helpful during some of the debug, but also very wordy for some stuff that uses 2k buffers. One thought would be to put it under a separate flag so it could be turned off (maybe there could be a buffer details flag)?

File: src/xdr_ioq.c
Comment (ffilz) [reply src/xdr_ioq.c:703]: Ah, good point. I'll move that out of the local variable definition, or make it a conditional assignment:

struct xdr_ioq_uv *next = TAILQ_NEXT(have, q) ? IOQ_(TAILQ_NEXT(have, q))  : NULL:

**Review by dang — COMMENTED** [review]
This presumably shouldn't be merged?  Is there outstanding work to do?

**Review by dang — APPROVED** [review]

**Pull request merged**
